### ISSUE TITLE
Syntax error on Pattern data type

### ISSUE THREAD

**Issue by fuero** [rodjek/puppet-lint#833]
*Update*: I guess it's connected to `(?x)`, as this works fine if the pattern is put on a single line.

Puppet-Lint chokes on the following type alias:

```
type Profiles::Ensure::Package = Variant[
  Enum['present', 'installed', 'absent', 'purged', 'held', 'latest'],
  # Fedora/EL Version scheme (NE)VRA
  Pattern[/(?x)
  ^
    (?:\d+\.)+\d+           # Regular version number part (0.1.0.0.x)
    -\d+                    # Release
    (?:\.[A-Za-z0-9]+)?     # custom suffix e.g. .20180917gitabcdef1
    (?:\.el\d(?:\.\d+)?)?   # distro suffix e.g. el7, el7.2
  $
  /]
]
```
Running the validator works without complaint, as do the unit tests I wrote for this.

```
$ puppet parser validate --verbose types/ensure/package.pp
$
```

Puppet-Lint does complain:
```
$ puppet-lint types/ensure/package.pp
ERROR: Syntax error on line 5
Try running `puppet parser validate <file>`
```

Versions:

```
$ puppet-lint --version
puppet-lint 2.3.6
$ puppet --version
5.5.6
$ ruby --version
ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux]
$ lsb_release -a
LSB Version:	1.4
Distributor ID:	Arch
Description:	Arch Linux
Release:	rolling
Codename:	n/a
```

**Comment by jcbollinger** [comment]
I have observed the same issue.  I'm using puppet-lint 2.3.6, as packaged (twice) in the Puppet Development Kit v1.7.  I concur that the problem seems to be associated with patterns that extend across multiple lines, as is enabled by turning on the `x` flag.

I encountered it with type definitions, just as reported above.  The affected types were accepted by several different revisions of Puppet 1.5.x, and they appear to have worked as intended, but puppet-lint rejected them altogether as containing syntax errors.

**Comment by jcbollinger** [comment]
Looking at the code, puppet-lint appears universally to reject regular expressions containing newlines (not just in the declarations of Pattern types).  Ruby and therefore Puppet unconditionally accept them, however, whether the `x` option is in effect or not.
